From 469631abe5b723ccabdcf9518bd0adf8fc767e89 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Wed, 13 Jan 2010 23:10:44 +0100
Subject: [PATCH] Make gconf/thumbnailer dependency optional

This just needs gconf-2.m4 to be present in m4/ directory in order to
run correctly.

Gentoo: https://bugs.gentoo.org/show_bug.cgi?id=289856
---
 configure.in |   39 +++++++++++++++++++++++++--------------
 1 files changed, 25 insertions(+), 14 deletions(-)

diff --git a/configure.in b/configure.in
index 73948f9..c4860d6 100644
--- a/configure.in
+++ b/configure.in
@@ -499,21 +499,32 @@ AM_CONDITIONAL(WITH_LIBGSF_BONOBO, test $have_bonobo = yes)
 
 # ===================
 # GConf configuration
+AC_ARG_WITH([gconf],
+    AC_HELP_STRING([--without-gconf],[disable the use of gconf (default: auto)]),
+    [],
+    [with_gconf=auto])
+
+AS_IF([test "$with_gconf" != "no"], [
+    PKG_CHECK_EXISTS([gconf-2.0 >= 2.24.0],[
+        AC_CHECK_PROG([GCONFTOOL], [gconftool-2], [gconftool-2], [no])
+        AS_IF([test "$GCONFTOOL" = "no"], [
+            AC_MSG_WARN([thumbnailer will not be built, unable to find gconftool-2])
+            have_gconf=no
+            ], [
+            have_gconf=yes
+            ])
+        ],[have_gconf=no])
+    ], [have_gconf=no])
+
+AS_IF([test "$have_gconf" = "yes"], [], [
+    AS_IF([test "$with_gconf" = "yes"],
+        [AC_MSG_ERROR([gconf support requested, but not available.])])
+    ])
+
+AM_GCONF_SOURCE_2
+AM_CONDITIONAL(WITH_GCONF, test "$have_gconf" = "yes")
 
-dnl The following conditional is set in AM_GCONF_SOURCE_2.
-dnl Because we may skip its execution, we have to set a default here.
-AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL], [false])
-
-dnl Don't publish the GCONFTOOL variable, AM_GCONF_SOURCE_2 has ``gconftool-2'' hardwired.
-m4_pattern_allow([^AM_GCONF_SOURCE_2$])
-GCONFTOOL=
-AC_CHECK_PROG([GCONFTOOL], [gconftool-2], [gconftool-2], [no])
-if test "x$GCONFTOOL" = "xno"; then
-    AC_MSG_WARN([thumbnailer will not be built, unable to find gconftool-2])
-else
-    AM_GCONF_SOURCE_2
-fi
-AM_CONDITIONAL(WITH_GCONF, test "x$GCONFTOOL" != "xno")
+# ===================
 
 LDFLAGS="-no-undefined $LDFLAGS"
 
-- 
1.6.6.rc3

